protocol/auditlog.go

```diff
@@ -100,6 +116,7 @@ func (l ConiksAuditLog) Insert(addr string, signKey sign.PublicKey,
 	// create the new directory history
 	h = newDirectoryHistory(addr, signKey, snaps[0])
```

Comment:
Similar to my comment in https://github.com/coniks-sys/coniks-go/pull/170#discussion_r132728501, I don't think that it's `newDirectoryHistory`'s job to verify `snaps[0]`, especially if it's being called in the context of `Audit()`.